src/vs/workbench/services/textfile/browser/textFileService.ts

```diff
@@ -462,7 +462,9 @@ export abstract class AbstractTextFileService extends Disposable implements ITex
 	}
 
 	suggestFilename(mode: string, untitledName: string) {
-		const extension = this.modeService.getExtensions(mode)[0];
+		// {{SQL Carbon Edit}} - pass language not mode for getting the extension.
+		const langName = this.modeService.getLanguageName(mode);
+		const extension = langName ? this.modeService.getExtensions(langName)[0] : undefined;
```

Comment:
do we know that `this.modeService.getExtensions` always returns an array with at least 1 element?  Or should we null check before accessing?

Reply:
it can return an empty array but it's the same with the previous call (const extension = this.modeService.getExtensions(mode)[0]) and I am trying to stick to the pattern they followed. If you think it's good to add the null check, I can add it but do I need to update the other statements in this method that have the same issue?